packages/monitor-v2/src/monitor-polymarket/MonitorProposalsOrderBook.ts

```diff
@@ -50,22 +50,18 @@ export async function monitorTransactionsProposedOrderBook(
   const markets = await getPolymarketMarkets(params);
 
   const marketsWithAncillary = await getMarketsAncillary(params, markets);
+
   // Filter out markets that do not have a proposal event.
   const marketsWithEventData: PolymarketWithEventData[] = marketsWithAncillary
-    .filter((market) => proposalEvents.find((event) => event.ancillaryData === market.ancillaryData))
+    .filter((market) =>
+      proposalEvents.find(
+        (event) => event.ancillaryData === market.ancillaryData && event.timestamp === market.requestTimestamp
+      )
+    )
     .map((market) => {
-      const events = proposalEvents.filter((event) => event.ancillaryData === market.ancillaryData);
-      // get last proposal event
-      const event = events.reduce((maxEvent, currentEvent) => {
-        if (currentEvent.eventBlockNumber > maxEvent.eventBlockNumber) {
-          return currentEvent;
-        } else if (currentEvent.eventBlockNumber === maxEvent.eventBlockNumber) {
-          if (currentEvent.eventIndex > maxEvent.eventIndex) {
-            return currentEvent;
-          }
-        }
-        return maxEvent;
-      });
+      const event = proposalEvents.find(
```

Comment:
We don't need to find the proposal with highest timestamp anymore, this was a workaround to handle the situation when a proposed request was disputed, re-requested and proposed again.

As we know have both the `ancillaryData` and `requestTimestamp` fetched from the [UmaConditionalTokensBinaryAdapter](https://polygonscan.com/address/0xCB1822859cEF82Cd2Eb4E6276C7916e692995130) or [UmaCtfAdapter](https://polygonscan.com/address/0x6a9d222616c90fca5754cd1333cfd9b7fb6a4f74) we can uniquely find the latest and relevant proposal corresponding to a market

Reply:
Perfect, so if we know the identifier, timestamp, requesting address, and ancillary data match, it's impossible for this to be the wrong request because these are the variables the OOV2 uses to uniquely identify a request, right?

Reply:
Agree!